### DISCUSSION TITLE
Syncing state between browser tabs

### DISCUSSION THREAD

**Discussion by mbajur** [joshleblanc/view_component_reflex#69] [Q&A]
Hello!
I'm trying to implement a scenario in which interacting with component in one browser window triggers the component reload in all the other opened browser windows.

Let's say there is a "No pages yet - [create one]" widget/component visible in CMS admin dashboard page. When user A clicks on "create one" button, reflex does it's magic, creates a page and reloads the component both in the actor window AND in all the other actors windows  so everyone has the most up-to-date version of the component visible on their screens. 

I'm pretty sure that's not possible with current implementation of ViewComponentReflex but i might be wrong :) Also, maybe there is some smart way of doing that?

Thanks!

**Comment by joshleblanc** [comment]
Hi @mbajur, if you pass `isolate: false` to the `StimulusReflex.initialize` call, it should sync the state between tabs.

**Reply by mbajur** [reply]
Yes, but i presume it's gonna sync the state only for the user who interacted with the component? Not all the users that are currently seeing the component?

I'm rather looking for a way to trigger component reload for all the user, not just the one interacting.

**Reply by joshleblanc** [reply]
Can you try using the `stream_to` method within the reflex to specify a custom channel that will broadcast to all users?

Something like 

```ruby
def create_page
  stream_to EveryoneChannel
 
  # do stuff
end
```

I can test this myself in a few hours. I know I implemented something like this a few months back, just a little fuzzy on the details. I can take a closer look after work.

**Comment by joshleblanc** [comment]
@mbajur Looks like we can do this without too much trouble. 

Create a new channel for global messages, `rails g channel Global`

```ruby
class GlobalChannel < ApplicationCable::Channel
  def subscribed
    stream_from "global"
  end
end
```

Run the data through cableready

```js
import consumer from "./consumer"
import CableReady from 'cable_ready';

consumer.subscriptions.create("GlobalChannel", {
  received(data) {
    if (data.cableReady) CableReady.perform(data.operations)
  }
});
```

Then bypass VCR's default morph entirely.

```ruby
def my_reflex
  prevent_refresh!
  stream_to "global"
  
  @count = @count + 1
  
  refresh!
end
```

I tested this with view_component_reflex_expo, and it's working fine. Let me know if you run into any issues.
